apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Tests/AggregateFieldTest.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using System;
+using Xunit;
+using static Google.Cloud.Firestore.AggregateField;
+using static Google.Cloud.Firestore.V1.StructuredAggregationQuery.Types;
+using static Google.Cloud.Firestore.V1.StructuredAggregationQuery.Types.Aggregation.Types;
+
+namespace Google.Cloud.Firestore.Tests;
+public class AggregateFieldTest
+{
+    [Fact]
+    public void NullFieldCheck()
+    {        
+        Assert.Throws<ArgumentNullException>(() => Sum(null));
+        Assert.Throws<ArgumentNullException>(() => Average(null));
+    }
+
+    [Fact]
+    public void AggregateFieldTests()
+    {
+        // Alias generated by client library
+        Assert.Equal(new Aggregation { Count = new Count(), Alias = CountAlias }, Count().Aggregation);
+        Assert.Equal(new Aggregation { Sum = new Sum { Field = FieldPath.FromDotSeparatedString("age").ToFieldReference() }, Alias = "Sum_age" }, Sum("age").Aggregation);
```

Comment:
Rather than copying the implementation code, I think it would be better for the expectation to be "raw" - so `Field = new FieldReference { FieldPath = "age" }` etc.